examples/tf-azurerm-computegroup/main.tf

```diff
@@ -0,0 +1,122 @@
+  location = "${var.location}"
+  tags     = "${var.tags}"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "acctvn"
+  address_space       = ["10.0.0.0/16"]
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+}
+
+resource "azurerm_subnet" "subnet1" {
+  name                 = "acctsub"
+  resource_group_name  = "${azurerm_resource_group.rg.name}"
+  virtual_network_name = "${azurerm_virtual_network.vnet.name}"
+  address_prefix       = "10.0.1.0/24"
+}
+
+resource "azurerm_public_ip" "publicip" {
+  name                         = "test"
+  location                     = "${var.location}"
+  resource_group_name          = "${azurerm_resource_group.rg.name}"
+  public_ip_address_allocation = "static"
+  domain_name_label            = "${azurerm_resource_group.rg.name}"
+  tags                         = "${var.tags}"
+}
+
+resource "azurerm_lb" "loadbalancer" {
+  name                = "loadbalancer"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  tags                = "${var.tags}"
+
+  frontend_ip_configuration {
+    name                 = "PublicIPAddress"
+    public_ip_address_id = "${azurerm_public_ip.publicip.id}"
+  }
+}
+
+resource "azurerm_lb_backend_address_pool" "bpepool" {
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  loadbalancer_id     = "${azurerm_lb.loadbalancer.id}"
+  name                = "BackEndAddressPool"
+}
+
+resource "azurerm_lb_nat_pool" "lbnatpool" {
+  count                          = "${length(var.protocol)}"
+ resource_group_name            = "${azurerm_resource_group.rg.name}"
+ name                           = "${element(keys(var.protocol), count.index)}"
+ loadbalancer_id                = "${azurerm_lb.loadbalancer.id}"
+ protocol                       = "${element(var.protocol["${element(keys(var.protocol), count.index)}"], 1)}"
+ frontend_port_start            = "${element(var.protocol["${element(keys(var.protocol), count.index)}"], 2)}"
+ frontend_port_end              = "${element(var.protocol["${element(keys(var.protocol), count.index)}"], 3)}"
+ backend_port                   = "${element(var.protocol["${element(keys(var.protocol), count.index)}"], 0)}"
+ frontend_ip_configuration_name = "PublicIPAddress"
+}
+
+resource "azurerm_virtual_machine_scale_set" "vmss" {
+  name                = "vmscaleset"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  upgrade_policy_mode = "Manual"
+  tags                = "${var.tags}"
+
+  sku {
+    name     = "${var.vm_size}"
+    tier     = "Standard"
+    capacity = "${var.nb_instance}"
+  }
+
+  storage_profile_image_reference {
+    id        = "${var.os["id"]}"
+    publisher = "${var.os["publisher"]}"
+    offer     = "${var.os["offer"]}"
+    sku       = "${var.os["sku"]}"
+    version   = "${var.os["version"]}"
```

Comment:
I'd argue these should be separate variables, rather than in the hash - as it makes it possible for Terraform to validate them and prompt for them if they're not specified